clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/model/EcsLoadBalancerCache.java

```diff
@@ -54,6 +54,6 @@ public String getName() {
 
   @Override
   public String getType() {
-    return loadBalancerType;
+    return cloudProvider;
```

Comment:
Is there a reason why type returns the ecscloudprovider id? Shouldn't the getCloudProvider call be used instead?

Reply:
> Shouldn't the getCloudProvider call be used instead?

it should! `getType` was actually [deprecated](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/LoadBalancer.java#L52) in favor of `getCloudProvider`, however it's still a part of the the `LoadBalancer` interface which is why the method is still here.

Returning `cloudProvider` instead of `loadBalancerType` I think aligns it better to it's intended purpose (plus `loadBalancerType` should always be "application" in the case of ECS, vs. AWS LBs, which is where we're sourcing the cache data), however @clareliguori might know more.

Reply:
The @Data annotation on the class should already provide the default setters and getters. In this case, I don't think you'll need to override any of the accessors (as long as you update the methods calling getType to call getCloudProvider.

Reply:
there currently isn't a field just called "type", so we need the override method regardless, and I would rather return the cloud provider ID since this is in line with how [most](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/model/AmazonLoadBalancer.groovy#L39) [other](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/model/AppengineLoadBalancer.groovy#L38) [providers](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/model/CloudFoundryLoadBalancer.java#L111) are currently setting this field.

Reply:
As discussed offline, this makes sense in order to maintain consistency with the other providers. Since the interface only has a getName and getType method, it would potentially break in many places if we add logic to create a new getProvider method.